arch/posix/CMakeLists.txt

```
@@ -64,7 +64,14 @@ else() # Linux.x86_64
     check_set_compiler_property(APPEND PROPERTY fpsse2 "SHELL:-msse2 -mfpmath=sse")
     zephyr_compile_options($<TARGET_PROPERTY:compiler,fpsse2>)
     target_compile_options(native_simulator INTERFACE "$<TARGET_PROPERTY:compiler,fpsse2>")
-  endif ()
+
+    # Let's request from the host C library (glibc) that time_t be a 64bit type for x86 32bit builds
+    # so we avoid the year 2038 issue.
+    # https://sourceware.org/glibc/wiki/Y2038ProofnessDesign
+    if (CONFIG_ARCH_POSIX_FORCE_64BIT_TIME_T AND CONFIG_EXTERNAL_LIBC)
```

Would something like the following work? That would eliminate the need to define an additional Kconfig.

```suggestion
    if((NOT CONFIG_LIBC_ALLOW_32_BIT_TIME) AND CONFIG_EXTERNAL_LIBC)
```

---

Not in a good way.
CONFIG_ARCH_POSIX_FORCE_64BIT_TIME_T is a dangerous option that deserves the description/warning it comes with, and which will not work with older glibCs.
while LIBC_ALLOW_LESS_THAN_64BIT_TIME just disables a build check.